### PR TITLE
Address feedback on RFC 0002 (Authentication Framework)

### PR DESCRIPTION
**Pull request by kylef**
@pksunkara This pull request is to address the feedback you've provided on #4 against RFC 0002.

Thanks for the feedback!

### CODE REVIEW

File: draft/0002-authentication.md [hunk 1]
```diff
@@ -172,7 +186,8 @@ can be marked as authenticated by placing an `Authenticated` list item.
 
 ### Actions
 
-You can mark a specific action as authenticated using the following:
+You can mark a specific action include all example requests as authenticated
```
Comment (pksunkara) [inline draft/0002-authentication.md]:
``` diff
- action include all example requests
+ action and include all example requests under it
```

File: draft/0002-authentication.md [hunk 2]
```diff
@@ -119,13 +127,18 @@ You specify that multiple authentication schemes are supported, using an
     + (Passphrase)
 ```
 
+If you do not need to override attributes, then you may use the shorthand equivilence:
```
Comment (pksunkara) [inline draft/0002-authentication.md]: equivalence

File: draft/0002-authentication.md [hunk 3]
```diff
@@ -54,8 +54,16 @@ following in an API Blueprint:
 **NOTE**: *The basic scheme is used as an example in this RFC, and is not
 defined within this RFC.*
 
-Under the authentication scheme heading, you may define properties within a
-list item. An authentication scheme may provide a set of properties or defaults.
+Under the authentication scheme heading, you may attach a description and
+define properties within a list item. An authentication scheme may
+provide a set of properties or defaults.
```
Comment (pksunkara) [inline draft/0002-authentication.md]: Don't you mean `properties with defaults`?

File: draft/0002-authentication.md [hunk 4]
```diff
@@ -54,8 +54,16 @@ following in an API Blueprint:
 **NOTE**: *The basic scheme is used as an example in this RFC, and is not
 defined within this RFC.*
 
-Under the authentication scheme heading, you may define properties within a
-list item. An authentication scheme may provide a set of properties or defaults.
+Under the authentication scheme heading, you may attach a description and
+define properties within a list item. An authentication scheme may
```
Comment (pksunkara) [inline draft/0002-authentication.md]: Each property is a list item. `properties within a list item` sounds different.

File: draft/0002-authentication.md (hunk 2 above)
Comment (pksunkara) [inline draft/0002-authentication.md]: `override properties`

File: draft/0002-authentication.md [hunk 5]
```diff
@@ -172,7 +186,8 @@ can be marked as authenticated by placing an `Authenticated` list item.
 
 ### Actions
 
-You can mark a specific action as authenticated using the following:
+You can mark a specific action include all example requests as authenticated
+using the following:
```
Comment (pksunkara) [inline draft/0002-authentication.md]: Actually this description part is a different format compared to the descriptions under `Resource` and `Resource groups`.

This has `You can mark` (active form), while they have `can be marked` (passive form)

**Comment by pksunkara** [comment]
How does the description work in Authentication Scheme?

``` apib
### Passphrase (Basic)

Polls API supports the Basic authentication scheme used with
pre-existing user accounts.

+ username: kyle
+ password: b2952d03bda09cb5f63b0162fbbee77c

+ Response 203 (application/json)

        {}
```

or like this:

``` apib
### Passphrase (Basic)

Polls API supports the Basic authentication scheme used with
pre-existing user accounts.

#### Properties
+ username: kyle
+ password: b2952d03bda09cb5f63b0162fbbee77c

+ Response 203 (application/json)

        {}
```

@zdne Might need your opinion on this.

---

**EDIT**: The previous code sample before the edit is as follows:

``` apib
### Passphrase (Basic)

Polls API supports the Basic authentication scheme used with
pre-existing user accounts.

+ Properties
    + username: kyle
    + password: b2952d03bda09cb5f63b0162fbbee77c

    + Response 203 (application/json)

            {}
```

**Comment by kylef** [comment]
@pksunkara I think the syntax for description should be as it is for [MSON Type Sections](https://github.com/apiaryio/mson/blob/master/MSON%20Specification.md#4-type-sections) for consistency. I've mentioned this in 3dd55a7. So:

``` apib
### Passphrase (Basic)

Polls API supports the Basic authentication scheme used with
pre-existing user accounts.

### Properties
+ username: kyle
+ password: b2952d03bda09cb5f63b0162fbbee77c

+ Response 203 (application/json)

            {}
```

I don't think responses should belong in properties, having a think on where and how to include responses.

**Comment by pksunkara** [comment]
> I think the syntax for description should be as it is for MSON Type Sections for consistency

Yeah, I agree and this is what I was talking about with my previous comment. I just mistakenly put `Properties` as a list item in sample code I gave.

> I don't think responses should belong in properties, having a think on where and how to include responses.

And now, we reached the issue I actually wanted to point out since the beginning of the PR. When @zdne used something like this, I was thinking about finding another way to do this. After seeing this in the PR, I was struggling to come up with a way to explain why this is not good. Looks like slowly going through the complete format of the Authentication Scheme was a good idea :smile:

File: draft/0002-authentication.md [hunk 6]
```diff
@@ -119,13 +129,18 @@ You specify that multiple authentication schemes are supported, using an
...
+```
+
 ### Resource Groups
 
 The authenticated keyword may be used within a resource group to indicate that
-all their children resources, actions and example requests inside the
-resource group support the specified authentication schemes. Any
-specific resource, action, or example requests may overide the inherited
-authentication schemes.
+all child resources, actions and example requests inside the resource group
```
Comment (zdne) [inline draft/0002-authentication.md]: nit: I would omit the word "example" altogether

**Comment by zdne** [comment]
:+1: 

Once comment tho. I would suggest using MSON / Attributes exclusively instead of JSONs. Not just in this PR but everywhere else (parameters RFC PR for example)

**Comment by pksunkara** [comment]
@kylef Could you add this as a card in trello so that we can finalise this soon?

**Comment by kylef** [comment]
I've addressed all the feedback, @pksunkara can you review and merge this please.

File: draft/0002-authentication.md [hunk 7]
```diff
@@ -119,13 +129,18 @@ You specify that multiple authentication schemes are supported, using an
...
+```
+
 ### Resource Groups
 
 The authenticated keyword may be used within a resource group to indicate that
-all their children resources, actions and example requests inside the
-resource group support the specified authentication schemes. Any
-specific resource, action, or example requests may overide the inherited
-authentication schemes.
+all child resources, actions and requests inside the resource group support
```
Comment (pksunkara) [inline draft/0002-authentication.md]: `example requests`

File: draft/0002-authentication.md [hunk 8]
```diff
@@ -119,13 +129,18 @@ You specify that multiple authentication schemes are supported, using an
...
+the specified authentication schemes. Any specific resource, action, or
+requests may override the inherited authentication schemes.
```
Comment (pksunkara) [inline draft/0002-authentication.md]: `example requests`. We have been calling them that throughout the RFCs.

File: draft/0002-authentication.md [hunk 9]
```diff
@@ -153,8 +168,9 @@ Resource groups may be marked as authenticated using the following:
 
 ### Resources
 
-Similiary to resource groups, a resource and it's children actions and examples
-can be marked as authenticated by placing an `Authenticated` list item.
+Similiary to resource groups, a resource and it's children actions and
```
Comment (pksunkara) [inline draft/0002-authentication.md]: Now that I think about it, The phrase `Similiary to resource groups` sounds a bit weird.

File: draft/0002-authentication.md [hunk 10]
```diff
@@ -153,8 +168,9 @@ Resource groups may be marked as authenticated using the following:
...
+example requests may be marked as authenticated by placing an
```
Comment (pksunkara) [inline draft/0002-authentication.md]: Please omit the word `example` in all other places too.

Comment (pksunkara) [inline draft/0002-authentication.md]: `using the `Authenticated` keyword as follows:`

File: draft/0002-authentication.md (hunk 8 above)
Comment (pksunkara) [inline draft/0002-authentication.md]: `request`.

File: draft/0002-authentication.md (hunk 9 above)
Comment (pksunkara) [inline draft/0002-authentication.md]: How about this?

```
Resources, including all the actions and requests within the resource may be marked as authenticated using the  `Authenticated` keyword as follows:
```

**Comment by pksunkara** [comment]
> I don't think responses should belong in properties, having a think on where and how to include responses.

I still want to do this. Do you have any ideas?

**Pull request merged**